### ISSUE TITLE
TextInput font size is inadequate on iOS

### ISSUE THREAD

**Issue by FrivalszkyP** [carbon-design-system/carbon#2019]
## Environment

> Operating system

Any recent iOS versions

> Browser

Safari on iOS 

## Detailed description

> What version of the Carbon Design System are you using?

carbon-components@9.82.13 is in use. 

> What did you expect to happen? 

To be clear, the behavior is expected and it is a feature rather than a bug. It's more like a cosmetic issue. It's not unreasonable to use at least a 16px font size by default in our forms in the Carbon Design System. It is better for overall accessibility on any mobile device.

The zoom in behavior on iOS is a nice feature for smaller input fields, but it can result in a zoomed state that hides other important information surrounding the input field. Moreover, if you submit the form or navigate away, the browser stays zoomed, which is quite annoying.

So what I expected is for the browser not to zoom in when I focus an input field, and for the input field to have at least a 16-pixel font size on mobile.

> What happened instead?

The browser zoomed in and stayed that way even after navigating to another page.

> What WCAG 2.1 checkpoint does the issue violate?

Success Criterion 1.4.3 Contrast (Minimum)

## Steps to reproduce the issue

1. Create a responsive webpage with a `<meta name="viewport" content="width=device-width, initial-scale=1">` in its HEAD section.
2. Include a text input field with a `bx--text-input` class on it.
3. Open the page on an iOS device in Safari
4. Tap the input field and see the browser get zoomed in.

https://codesandbox.io/embed/w72kk6nmr8?fontsize=14&hidenavigation=1&view=preview

## Additional information

One of these screenshots is taken on iOS without the input field in focus. The other is taken with the input field in focus. See the difference: the browser zoomed in on the input field.

<img width="490" alt="screenshot 2019-03-08 at 11 24 32" src="https://user-images.githubusercontent.com/3766124/54023369-159c2d00-4195-11e9-8e6e-730d03717121.png">
<img width="490" alt="screenshot 2019-03-08 at 11 24 42" src="https://user-images.githubusercontent.com/3766124/54023371-159c2d00-4195-11e9-87ea-e716f4c462d4.png">

**Comment by dakahn** [comment]
Hey thanks for submitting and issue. Let me make sure I have this right

- _you believe the default font-size on the text field component is an accessibility concern_
- _you don't want your zoomed in state to carry over between pages_

Setting a minimum font size (lets say 18px) and declaring it "accessible" or "more accessible" is a tough to do. There's no standard mobile aspect ratio so understanding whether the type on your site/app is accessible and usable requires thorough testing. Smaller fonts could be more accessible than larger fonts given contrast ratios, environment etc could be markedly less accessible.

An affordance to mitigate this is allowing your users to zoom in and maintaining that zoom level across screens.

Let me know If I'm not quite getting what you're saying. 👍🏽
